charger/zaptec.go

```diff
@@ -275,31 +302,57 @@ var _ api.PhaseSwitcher = (*Zaptec)(nil)
 
 // Phases1p3p implements the api.ChargePhases interface
 func (c *Zaptec) Phases1p3p(phases int) error {
-	res, err := c.statusG.Get()
-
-	if err == nil {
-		data := zaptec.Update{
-			MaxChargePhases: &phases,
-		}
-
-		err = c.chargerUpdate(data)
+	var err error
+	if c.version == zaptec.ZaptecGo2 {
+		err = c.switchForZaptecGo2(phases)
+	} else {
+		err = c.switchForZaptecPro(phases)
 	}
 
 	if err == nil && c.priority {
 		data := zaptec.SessionPriority{
 			PrioritizedPhases: &phases,
 		}
 
+		res, err := c.statusG.Get()
+		if err != nil {
+			return err
+		}
+
 		if session := res.ObservationByID(zaptec.SessionIdentifier); session != nil {
 			err = c.sessionPriority(session.ValueAsString, data)
 		} else {
 			err = errors.New("unknown session")
 		}
 	}
-
 	return err
 }
 
+func (c *Zaptec) switchForZaptecPro(phases int) error {
+	data := zaptec.Update{
+		MaxChargePhases: &phases,
+	}
+	return c.chargerUpdate(data)
+}
+
+func (c *Zaptec) switchForZaptecGo2(phases int) error {
+	curr := 32
```

Comment:
Ideally this should be the installation default already set, not to burn something. Could we get this somewhere on startup?

Reply:
availableCurrent limits itself always to MaxCurrent. 
but agree... would be cleaner to read MaxCurrent at startup